enhancement: strokes on buttons and cards

nakedfool opened the pull request:
Closes #3337

Check out these changes on all preview types:

[Title preview](https://jokerace-git-enhancement-strokes-on-buttons-and-cards-jokerace.vercel.app/contest/basetestnet/0x2c87ab379c27084470c654e4160098e8aef109ca)
[Image preview](https://jokerace-git-enhancement-strokes-on-buttons-and-cards-jokerace.vercel.app/contest/basetestnet/0xa12ad64a32cb3859eb3c3ee5d1362574ef7f96a9)
[Tweet preview](https://jokerace-git-enhancement-strokes-on-buttons-and-cards-jokerace.vercel.app/contest/basetestnet/0x77ef7449647d4ec4647f061335406845fe129f68)

divine-economy reviewed (changes requested):
two thoughts here @nakedfool :

1. the color button in the preview for title contests looks a little tight. i don't think it's too small since it looks good on the other contest pages, but it might just be too close to the carrot icon (v). maybe we move it another 8px to the left to give a bit more space?

2. i think we need some hover states for these buttons -- i apologize for not mocking that up. for the full-color preview button, maybe it could be embossed (ie comes up and to the right with a drop shadow on the lower left) or just have a stroke (DEDAD6)? for the button with the rainbow stroke, maybe it could become a full rainbow button when you hover, similar to the other one?  let me know if you need me to mock these up.

3. how do we feel about the double stroke for the button that you see when you expand text? glad to just do one stroke instead of two if we think that's better too 

4. i don't feel strongly, but i hadn't mocked up the comments button for previews of text contests on mobile, and we don't have it appearing on the preview for images/tweets contests either. but if you'd prefer to keep it here @nakedfool (i see you included it in mobile version of text previews), i'm glad to defer to you!

nakedfool commented:
> two thoughts here @nakedfool :
> 
> 1. the color button in the preview for title contests looks a little tight. i don't think it's too small since it looks good on the other contest pages, but it might just be too close to the carrot icon (v). maybe we move it another 8px to the left to give a bit more space?
> 2. i think we need some hover states for these buttons -- i apologize for not mocking that up. for the full-color preview button, maybe it could be embossed (ie comes up and to the right with a drop shadow on the lower left) or just have a stroke (DEDAD6)? for the button with the rainbow stroke, maybe it could become a full rainbow button when you hover, similar to the other one?  let me know if you need me to mock these up.
> 3. how do we feel about the double stroke for the button that you see when you expand text? glad to just do one stroke instead of two if we think that's better too
> 4. i don't feel strongly, but i hadn't mocked up the comments button for previews of text contests on mobile, and we don't have it appearing on the preview for images/tweets contests either. but if you'd prefer to keep it here @nakedfool (i see you included it in mobile version of text previews), i'm glad to defer to you!

1. Agreed, moved it another 8px, space between them is now 16px

2. Yeah animating linear gradient is very tricky (will be easier with motion library in the future) but i applied shadow animation for full-color preview button and added rainbow bg for the button with rainbow stroke

3. Ohh agreed, single rainbow border looks much better imo, nice call.

4. Yup it was there because of previous iterations, just removed it now.

divine-economy reviewed (approved):
love all these updates @nakedfool !!
